packages/odyssey-react-mui/src/test-selectors/querySelector.ts

```diff
@@ -11,188 +11,238 @@
+        >,
+      };
+
+      capturedElement =
+        featureTestSelector.selector.method === "ByRole"
+          ? getByRoleQuerySelector({
+              ...sharedProps,
+              role:
+                Array.isArray(featureTestSelector.selector.role) || role
+                  ? role || ""
+                  : featureTestSelector.selector.role,
+            })
+          : getByTextQuerySelector({
+              ...sharedProps,
+              selectionMethod: featureTestSelector.selector.method,
+              text: featureTestSelector.selector.text,
+            });
+    } else if (
+      "isControlledElement" in featureTestSelector &&
+      featureTestSelector.isControlledElement
+    ) {
+      try {
+        capturedElement = getControlledElement({ element: containerElement });
+      } catch (error) {
+        if (queryMethod === "query") {
+          capturedElement = null;
+        }
+
+        throw error;
+      }
+    }
+
+    if (!capturedElement) {
+      throw new ElementError(
+        "No child HTML element available",
+        containerElement,
+      );
+    }
+
+    if (!("accessibleText" in featureTestSelector)) {
+      throw new Error("Missing `accessibleText` in `FeatureTestSelector`");
+    }
+
+    const getAccessibleText = <
+      LabelName extends LocalFeatureTestSelector extends AccessibleLabelSelector
+        ? keyof LocalFeatureTestSelector["accessibleText"]
+        : keyof AccessibleLabelSelector,
+    >(
+      labelName: LabelName,
+    ) => {
+      if (!capturedElement) {
+        throw new ElementError(
+          "No child HTML element available",
+          containerElement,
+        );
+      }
+
+      return getComputedAccessibleText({
+        element: capturedElement,
+        type: featureTestSelector.accessibleText[labelName],
+      });
+    };
+
+    const selectChild = <
+      FeatureName extends LocalFeatureTestSelector extends FeatureSelector
+        ? keyof LocalFeatureTestSelector["feature"]
+        : keyof FeatureSelector,
+      ChildQueryMethod extends QueryMethod,
+    >(
+      childProps: {
+        featureName: FeatureName;
+      } & InnerQuerySelectorProps<
+        LocalFeatureTestSelector extends FeatureSelector
+          ? LocalFeatureTestSelector["feature"][FeatureName]
+          : FeatureTestSelector,
+        ChildQueryMethod
+      >,
+    ) => {
+      if (!capturedElement) {
+        throw new ElementError(
+          "No child HTML element available",
+          containerElement,
+        );
+      }
+
+      if (!("feature" in featureTestSelector)) {
+        throw new Error("Missing `feature` in `FeatureTestSelector`");
+      }
+
+      type Options = Record<
+        LocalFeatureTestSelector extends FeatureSelector
+          ? LocalFeatureTestSelector["feature"][FeatureName] extends TestSelector
+            ? keyof LocalFeatureTestSelector["feature"][FeatureName]["selector"]["options"]
+            : string
+          : string,
```

Comment:
Are these string cases expected? Or should they be `never`s?

Reply:
Good question.

`InnerQuerySelectorProps` doesn't return `{ options: something }` if you don't have options, so `string` or `never`, it doesn't matter. Regardless, I changed it to `never` since that's more accurate :thumbsup:.